platform/os_windows.go

```diff
@@ -189,6 +208,188 @@ func SetSdnRemoteArpMacAddress() error {
+	if err != nil {
+		return 0, err
+	}
+
+	intValue, err := strconv.Atoi(regvalue)
+	if err != nil {
+		return 0, fmt.Errorf("failed to convert PriorityVLANTag value to integer: %w", err)
+	}
+
+	return intValue, nil
+}
+
+// adapter is version 4 and up since adapter's advance property consists of reg key : PriorityVLANTag
+// set reg value for Priorityvlantag of adapater to 3 if not set already
+func setMellanoxPriorityVLANTagValueForV4(adapterName string) error {
+	currentVLANTagValue, err := getMellanoxPriorityVLANTagValueForV4(adapterName)
+	if err != nil {
+		return fmt.Errorf("Error while checking registery value for PriorityVLANTag for adapater, continuing to reset the value: %v", err)
+	}
+
+	if currentVLANTagValue == desiredRegValueForVLANTag {
+		log.Printf("Mellanox PriorityVLANTag is already set to %v, skipping reset", desiredRegValueForVLANTag)
+		return nil
+	}
+
+	cmd := fmt.Sprintf(
+		`Set-NetAdapterAdvancedProperty -Name "%s" -RegistryKeyword "%s" -RegistryValue %d`, adapterName, priorityVLANTagIdentifier, desiredRegValueForVLANTag)
+	_, err = ExecutePowershellCommand(cmd)
+	if err != nil {
+		return fmt.Errorf("error while setting up registery value for PriorityVLANTag for adapater: %w", err)
+	}
+
+	log.Printf("Successfully set Mellanox Network Adapter: %s with %s property value as %d", adapterName, priorityVLANTagIdentifier, desiredRegValueForVLANTag)
+	return nil
+}
+
+// Adapter is version 3 or less as PriorityVLANTag was not found in advanced properties of mellanox adpater
+func setMellanoxPriorityVLANTagValueForV3(adapterName string) error {
+	log.Printf("Searching through CIM instances for Network devices with %s in the name", mellanoxSearchString)
+	cmd := fmt.Sprintf(`Get-CimInstance -Namespace root/cimv2 -ClassName Win32_PNPEntity | Where-Object PNPClass -EQ "Net" | Where-Object { $_.Name -like "%s" } | Select-Object -ExpandProperty DeviceID`, mellanoxSearchString)
+	deviceid, err := ExecutePowershellCommand(cmd)
+
+	if err != nil {
+		return fmt.Errorf("error while executing powershell command to get device id of %s: %w", adapterName, err)
+	}
+	if deviceid == "" {
+		return fmt.Errorf("no network device found with %s in description", mellanoxSearchString)
+	}
+
+	log.Printf("Device ID found and Getting PNP device properites for %s", deviceid)
+	cmd = fmt.Sprintf(`Get-PnpDeviceProperty -InstanceId "%s" | Where-Object KeyName -EQ "DEVPKEY_Device_Driver" | Select-Object -ExpandProperty Data`, deviceid)
+	registryKeySuffix, err := ExecutePowershellCommand(cmd)
+	if err != nil {
+		return fmt.Errorf("error while executing powershell command to get registery suffix of device id %s: %w", deviceid, err)
+	}
+
+	registryKeyFullPath := registryKeyPrefix + registryKeySuffix
+
+	currentVLANTagValue, err := getMellanoxPriorityVLANTagValueForV3(registryKeyFullPath, adapterName)
+	if err != nil {
+		log.Errorf("Error while checking registery value for PriorityVLANTag for adapater, continuing to reset the value: %v", err)
```

Comment:
is there a reason to not continue on v4 if there is an error getting the value, but continue on v3?

Reply:
agree.. this should be return statement and should not log and continue... just like v4.. I am correcting this..